src/PoolManager.sol

```diff
@@ -219,13 +219,14 @@ contract PoolManager is IPoolManager, ProtocolFees, NoDelegateCall, ERC6909Claim
         uint256 feeForProtocol;
         uint24 swapFee;
         Pool.SwapState memory state;
-        (delta, feeForProtocol, swapFee, state) = pools[id].swap(
+        (delta, feeForProtocol, swapFee) = pools[id].swap(
```

Comment:
```suggestion
        (BalanceDelta delta, uint256 feeForProtocol, uint24 swapFee, Pool.SwapState memory state) = pools[id].swap(
```
That's a good find. Instead of making more intrusive changes, consider declaring types at assignment and returning `delta` explicitly while keeping the way you initialize `state` in `Pool.swap`.

Reply:
Yep really like this PR! But agree that this comment might help it read easier 🙏

Reply:
This option uses a little more gas, but it is more readable and understandable